### DISCUSSION TITLE
Java autoinstrumentation agent and MongoDB double traces

### DISCUSSION THREAD

**Discussion by dzydzany** [open-telemetry/opentelemetry-java-instrumentation#6384] [Q&A]
Hello, 

I am seeing double traces of MongoDB calls with different span ids and different execution times at one of our services. I am attaching the screenshot here. I am using version 1.16.0, and also I've tested with 1.15.0 These are configuration switches: 
              "-Dotel.exporter.otlp.endpoint=http://otel-collector:4317",
              "-Dotel.service.name=backend",
              "-javaagent:opentelemetry-javaagent.jar"

<img width="1322" alt="Screenshot 2022-07-29 at 11 18 31" src="https://user-images.githubusercontent.com/3818898/181729076-7010ab7c-8c6e-465c-b9bd-56ee93ce0f3d.png">

You can also notice for the other callbacks that all are double, just not expanded in the screenshot.

I saw a couple of issues that were resolved in past, and should already be in these agent versions... Can I pull more data for debugging somehow?

Thank you.

**Comment by mateuszrzeszutek** [comment]
Hey @dzydzany ,
On the first glance, this looks like a bug in the mongo client instrumentation. Could you [post an issue](https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/new/choose) in the instrumentation repo with some more details? E.g. what version of the mongodb client you're using, how you're calling it (e.g. whether you're using async or reactive methods), a code sample that's responsible for the faulty spans. It would be perfect if you could include a repro scenario (a minimal app or test) that recreates your issue.

**Comment by dzydzany** [comment]
Thank you very much, Mateusz. Let me see with my team what we can provide from all requested, and I will get back to you and convert this to an issue after it.
